src/bookshelf/utils.ts

```diff
@@ -123,10 +130,11 @@ function getAttrsList(data: Model): any {
  */
 function relationAllowed(bookOpts: BookOpts, relName: string): boolean {
   let { relations }: BookOpts = bookOpts;
+  let { fields }: RelationOpts = relations;
 
   return relations === true ||
-    (typeCheck('[String]', relations) &&
-      (relations as string[]).some((rel: string) => rel === relName));
+         relations instanceof Object ||
+         (fields instanceof Object && (fields as string[]).some((rel: string) => rel === relName));
```

Comment:
maybe test for `fields instanceof Array` instead? (I don't know if you would still need the casting `(fields as string[])`